### ISSUE TITLE
First starter help

### ISSUE THREAD

**Issue by pquerner** [phoice/phlexa-expressive-skeleton#5]
Hey,

I've just began working with this, but have some trouble.
After finally figuring out, that you DO NOT CALL https://example.com/hello/ but example.com/hello (missing slash at the end), else it will not fetch the correct config file (exception NULL applicationId) I get the next error:

```
{
    "error": "Invalid certificate to date"
}
```

As far as I can see this error is thrown from here:

```
object(Zend\Diactoros\Response\JsonResponse)#160 (9) {
["payload":"Zend\Diactoros\Response\JsonResponse":private]=>
array(1) {
["error"]=>
string(27) "Invalid certificate to date"
}
["encodingOptions":"Zend\Diactoros\Response\JsonResponse":private]=>
int(79)
["phrases":"Zend\Diactoros\Response":private]=>
array(66) {
[100]=>
string(8) "Continue"
[101]=>
string(19) "Switching Protocols"
[102]=>
string(10) "Processing"
[103]=>
string(11) "Early Hints"
[200]=>
string(2) "OK"
[201]=>
string(7) "Created"
[202]=>
string(8) "Accepted"
[203]=>
string(29) "Non-Authoritative Information"
[204]=>
string(10) "No Content"
[205]=>
string(13) "Reset Content"
[206]=>
string(15) "Partial Content"
[207]=>
string(12) "Multi-Status"
[208]=>
string(16) "Already Reported"
[226]=>
string(7) "IM Used"
[300]=>
string(16) "Multiple Choices"
[301]=>
string(17) "Moved Permanently"
[302]=>
string(5) "Found"
[303]=>
string(9) "See Other"
[304]=>
string(12) "Not Modified"
[305]=>
string(9) "Use Proxy"
[306]=>
string(12) "Switch Proxy"
[307]=>
string(18) "Temporary Redirect"
[308]=>
string(18) "Permanent Redirect"
[400]=>
string(11) "Bad Request"
[401]=>
string(12) "Unauthorized"
[402]=>
string(16) "Payment Required"
[403]=>
string(9) "Forbidden"
[404]=>
string(9) "Not Found"
[405]=>
string(18) "Method Not Allowed"
[406]=>
string(14) "Not Acceptable"
[407]=>
string(29) "Proxy Authentication Required"
[408]=>
string(15) "Request Timeout"
[409]=>
string(8) "Conflict"
[410]=>
string(4) "Gone"
[411]=>
string(15) "Length Required"
[412]=>
string(19) "Precondition Failed"
[413]=>
string(17) "Payload Too Large"
[414]=>
string(12) "URI Too Long"
[415]=>
string(22) "Unsupported Media Type"
[416]=>
string(21) "Range Not Satisfiable"
[417]=>
string(18) "Expectation Failed"
[418]=>
string(12) "I'm a teapot"
[421]=>
string(19) "Misdirected Request"
[422]=>
string(20) "Unprocessable Entity"
[423]=>
string(6) "Locked"
[424]=>
string(17) "Failed Dependency"
[425]=>
string(9) "Too Early"
[426]=>
string(16) "Upgrade Required"
[428]=>
string(21) "Precondition Required"
[429]=>
string(17) "Too Many Requests"
[431]=>
string(31) "Request Header Fields Too Large"
[444]=>
string(34) "Connection Closed Without Response"
[451]=>
string(29) "Unavailable For Legal Reasons"
[499]=>
string(21) "Client Closed Request"
[500]=>
string(21) "Internal Server Error"
[501]=>
string(15) "Not Implemented"
[502]=>
string(11) "Bad Gateway"
[503]=>
string(19) "Service Unavailable"
[504]=>
string(15) "Gateway Timeout"
[505]=>
string(26) "HTTP Version Not Supported"
[506]=>
string(23) "Variant Also Negotiates"
[507]=>
string(20) "Insufficient Storage"
[508]=>
string(13) "Loop Detected"
[510]=>
string(12) "Not Extended"
[511]=>
string(31) "Network Authentication Required"
[599]=>
string(29) "Network Connect Timeout Error"
}
["reasonPhrase":"Zend\Diactoros\Response":private]=>
string(11) "Bad Request"
["statusCode":"Zend\Diactoros\Response":private]=>
int(400)
["headers":protected]=>
array(1) {
["content-type"]=>
array(1) {
[0]=>
string(16) "application/json"
}
}
["headerNames":protected]=>
array(1) {
["content-type"]=>
string(12) "content-type"
}
["protocol":"Zend\Diactoros\Response":private]=>
string(3) "1.1"
["stream":"Zend\Diactoros\Response":private]=>
object(Zend\Diactoros\Stream)#158 (2) {
["resource":protected]=>
resource(242) of type (stream)
["stream":protected]=>
string(10) "php://temp"
}
}
LINE : 97 METHOD : Zend\HttpHandlerRunner\RequestHandlerRunner::run
```

Do you happen to know what I do wrong?
Also, can you explain what the HTTP Header "signature" means?

**Comment by pquerner** [comment]
Is this something about?

The request body?
```
  "request": {
    "type": "LaunchRequest",
    "requestId": "EdwRequestId.test",
    "locale": "de-DE",
    "timestamp": "2019-10-09T15:44:12Z" // <<<<<<<
  },
```

**Comment by pquerner** [comment]
I've disabled all checks in `\Phlexa\Request\Certificate\CertificateValidator::validate` because I think the cert (.pem file from amazon?) only lasted until 2017. All requests newer than this result in the error given above.

I only disabled them just to get started again.

**Comment by RalfEggert** [comment]
Hi @pquerner 

thanks for question. First of all, you need to pass the certification path in the request header for testing reasons. In my Postman I use these headers:

```
Content-Type: application/json
signaturecertchainurl: https://s3.amazonaws.com/echo.api/echo-api-cert-7.pem
signature: B/bxAdkIabkzsScfUsSfkz7pJrNLc1eoOOLk8qwG1ZudQRv7KcvyNa/91g74Zg3cRpifXEco4669MaZb4Cqs+vZ9TaTfftAMzy/Pc79AMuf1dU6GfUU7tp6cuavfqTD8cWlYN5TjEMCJbS1Y+VU929mX0edOZcZin7db6bOoZHu5gU8OSQ2r+6UMk88z5uuSjPyt9Du9vaC3Ics/Br30fEIplIgCt4y/UGRK76Rqo4L/DuNjty3o2mcU8bICK5xfZwCeH7b5UFwdjngtp8VfhKPtosZmCuWvMn+y9HoS06ll9cdI1FPLN9w7KwMZFY8UzTc+0GfAwntxzlowAwkPhQ==
```

If you want to disable the certification checks for testing you can do it in these files:

- /config/autoload/phlexa.config.development.php
- /config/autoload/phlexa.config.global.php

and then make sure that you have set write access for the `/data/cache` path.

Does that help?

**Comment by pquerner** [comment]
Well I used the same request header as you, yet still my verification fails due to the reason above. Can you confirm thats not an issue on your system?

**Comment by RalfEggert** [comment]
Ok, I work with this setup all day and cannot reproduce your issue so far.

Next question: Was the file `echo-api-cert-7.pem` downloaded and saved in the `/data/cache/` path?

**Comment by pquerner** [comment]
No, I did not download the file. It wasnt mentioned in the README.

**Comment by RalfEggert** [comment]
Sorry for the confusion. You should not download the file. The phoice framework will do this for you. Has an new `echo-api-cert-7.pem` been placed in the `/data/cache/` path when you got the error message. If not you should set write access to that path like I mentioned earlier.

Could you please check if you can find the file in the path?

**Comment by pquerner** [comment]
Sorry I have since then moved on to another strategy of developing Alexa Skills.

**Comment by RalfEggert** [comment]
Thanks @pquerner for the reply and for trying out phlexa. 

May I ask what your new strategy is?

**Comment by pquerner** [comment]
Using a different programming language. I tried out Node, then PHP and now I am using Java. Altough PHP is my main language I found it very hard for some reason to get started quickly on implementing actual logic to skills. Java is not my main language (neither is Node) but I found it quicker there.
Maybe when some times passes I'll give PHP another go. Its just talking JSON, can't be that hard to get up and running. ;)

Thanks tho for building this! I had great fun reading your blog posts and watching your videos / reading your online slides about this topic. :)
